packages/api-explorer/__tests__/lib/filter-empty-formdata.test.js

```diff
@@ -0,0 +1,230 @@
+import filterEmptyFormData from '../../src/lib/filter-empty-formdata'
+
+describe('filterEmptyFormData', () => {
+  it('filters all empty objects or objects with all undefined', () => {
+    const input = {
+      body: {
+        $set: {
+          position: [123, 456],
+          "image.$.replace": {
+            key: undefined,
+          },
+          "image.$.merge": {}
+        },
+        $unset: {
+          anotherKey: undefined
+        },
+        $inc: {},
+        $mul: {},
+        $currentDate: {},
+        $push: {
+          image: {}
+        }
+      }
+    }
+    const result = filterEmptyFormData(input)
+    expect(result).toEqual({
+      body: {
+        $set: { position: [123, 456] },
+      }
+    })
+  })
+  
+  it('filters objects into arrays', () => {
+    const input = {
+      "body": [{
+        "filter": {
+          "_st": "PUBLIC"
+        },
+        "update": {
+          "$set": {
+            "image.$.replace": {},
+            "image.$.merge": {}
+          },
+          "$unset": {},
+          "$inc": {},
+          "$mul": {},
+          "$currentDate": {},
+          "$push": {
+            "image": {}
+          }
+        }
+      }]
+    } 
+    const result = filterEmptyFormData(input)
+    expect(result).toEqual({
+      body: [{
+       filter: {_st: 'PUBLIC'} 
+      }]
+    })
+  })
+  
+  it('keeps unchanged arrays with null content', () => {
+    const input = { body: { posizione: [null, null], __STATE__: 'DRAFT'}}
+    const result = filterEmptyFormData(input)
+    expect(result).toEqual({
+      body: {
+        posizione: [null, null],
```

Comment:
'position' instead of 'posizione'

Reply:
renamed to a much more generic `someArray`.